### ISSUE TITLE
Documentation shouldn't inline the "__html taint" for dangerouslySetInnerHTML

### ISSUE THREAD

**Issue by WraithKenny** [facebook/react#4212]
On https://facebook.github.io/react/docs/tutorial.html (https://github.com/facebook/react/blob/master/docs/docs/tutorial.md), the Tutorial uses `<span dangerouslySetInnerHTML={{__html: rawMarkup}} />` though this is explicitly "recommend against" here: https://facebook.github.io/react/tips/dangerously-set-inner-html.html

**Comment by PiPeep** [comment]
It's recommended against *in most cases*. Here the tutorial explicitly shows that marked is used in a way that sanitizes the input, so react's own sanitation isn't needed.

This is a case where `dangerouslySetInnerHTML` is actually  the correct choice, and the tutorial tries to explain that. If it didn't use it, there's be no way to preserve the formatting generated in marked.

**Comment by jimfb** [comment]
The code is "recommended against" mainly because it explicitly bypasses security features (xss protection) provided by React.  The tutorial also explicitly calls out these security ramifications.  The point isn't that you should never use it (if the feature were inherently bad, we would remove it), the point is that you should be very careful and the decision to use `dangerouslySetInnerHTML` should be a conscious one.  Mentioning this feature in the tutorial isn't a bad thing, especially since people upgrading their legacy codebases (ie. people new to React) are the most likely to need this feature in order to interoperate with their existing codebase.

I'm going to go ahead and close this issue, because I think this was a conscious decision and not a bug.  Feel free to continue the discussion here, and if the community decides that hiding this feature is a good idea, we can re-open the issue to burry the feature deeper in the documentation :).

**Comment by WraithKenny** [comment]
I thought the note was about how to use it, `<div dangerouslySetInnerHTML={createMarkup()} />` not `<div dangerouslySetInnerHTML={{__html: getMarkup()}} />` in that the "type/taint" signature wasn't embedded in the element, as a check against the return 'type' (of sorts)...Having the `{{__html: variable}}` as an example in the tutorial seems to suggest a bad habit.

**Comment by jimfb** [comment]
Ah, so you're saying the "safe taint" should be moved up out of the jsx tag.  Yeah, I could get behind that :).

**Comment by WraithKenny** [comment]
I really wasn't commenting on the appropriateness of using `dangerouslySetInnerHTML` for all cases. Far from it, I was suggesting this (untested) change:
```
// tutorial7.js
var Comment = React.createClass({
  render: function() {
    var rawMarkup = {__html: marked(this.props.children.toString(), {sanitize: true})};
    return (
      <div className="comment">
        <h2 className="commentAuthor">
          {this.props.author}
        </h2>
        <span dangerouslySetInnerHTML={rawMarkup} />
      </div>
    );
  }
});
```

**Comment by WraithKenny** [comment]
(There are other examples of the usage in the various docs, but I lost track of them.)

**Comment by jimfb** [comment]
While we're fixing it, might as well do it correctly on the homepage too.

**Comment by berkerpeksag** [comment]
Looks like fixed in #4931. I think this can be closed now.
